library/src/scripts/flyouts/NewPostMenu.tsx

```diff
@@ -59,47 +71,285 @@ function ActionItem({ item, style }: { item: IAddPost; style: ITransition }) {
+                    dispatch({ type: "set_focused_item", item: undefined });
+                }
+                break;
+            default:
+                // We don't want to lose focus in case we already have
+                if (state.buttonFocus) {
+                    dispatch({ type: "set_button_focus", focus: true });
+                }
+                break;
+        }
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<any>) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        switch (event.key) {
+            case "Escape":
+                if (state.open) {
+                    dispatch({ type: "set_open", open: false });
+                    dispatch({ type: "set_button_focus", focus: true });
+                    dispatch({ type: "set_focused_item", item: undefined });
+                }
+                break;
+            case "Home":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: 0 });
+                }
+                break;
+            case "End":
+                if (state.open) {
+                    dispatch({ type: "set_focused_item", item: items.length - 1 });
+                }
+                break;
+            case "ArrowUp":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem + 1) % items.length });
+                }
+                break;
+            case "ArrowDown":
+                if (state.open && typeof state.focusedItem != "undefined") {
+                    dispatch({ type: "set_focused_item", item: (state.focusedItem - 1 + items.length) % items.length });
+                }
+                break;
+            case "Enter":
+                if (typeof state.focusedItem !== "undefined") {
+                    const item = items[state.focusedItem];
+                    switch (item.type) {
+                        case PostTypes.LINK:
+                            window.location.href = item.action as string;
+                            dispatch({ type: "set_open", open: false });
+                            dispatch({ type: "set_button_focus", focus: true });
+                            dispatch({ type: "set_focused_item", item: undefined });
+                            break;
+                        case PostTypes.BUTTON:
+                            (item.action as () => void)();
+                            dispatch({ type: "set_open", open: false });
+                            dispatch({ type: "set_button_focus", focus: true });
+                            dispatch({ type: "set_focused_item", item: undefined });
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                break;
+            default:
+                const itemList = items
```

Comment:
There's a bug here. If I open the menu with the keyboard, I am able to search for "t" to find the "to Google" link. However, after if i type "n" it should wrap around to the next item. 

If I'm on "New Poll", i should be able to hit "n" and jump to "New Idea". "n" again should jump to "New Discussion" and so on.

Reply:
I don't think it's a bug, if you press N it will work. I don't see the spec requiring case-insensitive. I'll talk to you more.

Reply:
Make it wrap, it feels weird to be "suck" on the bottom.

Reply:
I don't understand. I think it looks fine.